homeassistant/components/shelly/entity.py

```diff
@@ -535,35 +533,37 @@ def extra_state_attributes(self) -> dict[str, Any] | None:
 class ShellyRpcAttributeEntity(ShellyRpcEntity, entity.Entity):
     """Helper class to represent a rpc attribute."""
 
+    entity_description: RpcEntityDescription
+
     def __init__(
         self,
         wrapper: RpcDeviceWrapper,
         key: str,
         attribute: str,
-        description: RpcAttributeDescription,
+        description: RpcEntityDescription,
     ) -> None:
         """Initialize sensor."""
         super().__init__(wrapper, key)
-        self.sub_key = description.sub_key
         self.attribute = attribute
-        self.description = description
+        self.entity_description = description
 
         self._attr_unique_id = f"{super().unique_id}-{attribute}"
         self._attr_name = get_rpc_entity_name(wrapper.device, key, description.name)
-        self._attr_entity_registry_enabled_default = description.default_enabled
-        self._attr_device_class = description.device_class
-        self._attr_icon = description.icon
+        self._attr_should_poll = description.should_poll
```

Comment:
I found that `description.should_poll` (sometimes set to True) was never used for RPC sensors!
I am guessing it was a bug, and have now mapped it to `self._attr_should_poll`

Reply:
This is actually not a bug, We recently found that the device does not push values for some of the sensors, to resolve it we use two `DataUpdateCoordinator` classes:
- `RpcDeviceWrapper` - Main device coordinator which is using push updates
- `RpcPollingWrapper` - A pooling coordinator enabled only if one of the polling sensors is enabled and is using polling from the device.

When using `DataUpdateCoordinator` the `should_poll` attribute of the entity should be set to `False` regardless if based on polling since the `DataUpdateCoordinator` is handling the polling if needed.

The problem here is that I used a similar name to indicate which type of coordinator to use. It is not an entity attribute and should not be set to the entity (unless you have a better suggestion how to use a single coordinator for push and pool). We might need to name it better not to cause a confusion with the entity `should_poll` attribute.

This is where we assign a different coordinator by this flag:
https://github.com/home-assistant/core/blob/c202b24caeb0703611f3fc85145774c7d837be5d/homeassistant/components/shelly/entity.py#L191-L197